notifier/notifier.go

```diff
@@ -383,25 +391,25 @@ func (n *Notifier) Stop() {
 
 // Describe implements prometheus.Collector.
 func (n *Notifier) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
Actually, the notifier constructor should accept a `prometheus.Registerer` and metrics against that one.
See here for the global-free pattern: https://github.com/fabxc/tsdb/blob/master/db.go#L93-L169